src/core/sys/posix/netdb.d

```diff
@@ -594,6 +594,107 @@ else version( OpenBSD )
     enum EAI_PROTOCOL       = -13;
     enum EAI_OVERFLOW       = -14;
 }
+else version( DragonFlyBSD )
+{
+    /*
+     * Error return codes from gethostbyname() and gethostbyaddr()
+     * (left in h_errno).
+     */
+    struct hostent
+    {
+        char*   h_name;
+        char**  h_aliases;
+        int     h_addrtype;
+        int     h_length;
+        char**  h_addr_list;
+        extern (D) char* h_addr() @property { return h_addr_list[0]; } // non-standard
+    }
+
+    struct netent
+    {
+        char*   n_name;
+        char**  n_aliases;
+        int     n_addrtype;
+        uint32_t n_net;
+    }
+
+    struct protoent
+    {
+        char*   p_name;
+        char**  p_aliases;
+        int     p_proto;
+    }
+
+    struct servent
+    {
+        char*   s_name;
+        char**  s_aliases;
+        int     s_port;
+        char*   s_proto;
+    }
+
+    struct addrinfo
+    {
+        int         ai_flags;
+        int         ai_family;
+        int         ai_socktype = SOCK_STREAM;           /* socktype default value required to be able to perform getAddrInfo on DragonFlyBSD
```

Comment:
Fixed getAddrInfo issue in phobos/std/socket.d by giving addrinfo.ai_socktype a default value of SOCK_STREAM.
(Without it getAddrInfo() will generate an error:'servname not supported for ai_socktype' when hints.ai_socktype is not set.)

Reply:
Would it not be better to change `std.socket` instead?

Reply:
You mean to set the default value for ai_socktype, right ?

I did try to find a good location in std.socket first (When the unittest failed and pointed out the issue). I tried this, and it worked, but i did not like this solution very much. Maybe mostly caused by the location i choose to 'fix' the issue.
```diff
diff --git a/std/socket.d b/std/socket.d
index 361ec73da..ba68e1738 100644
--- a/std/socket.d
+++ b/std/socket.d
@@ -963,6 +963,13 @@ AddressInfo[] getAddressInfo(T...)(in char[] node, T options)
         else
             static assert(0, "Unknown getAddressInfo option type: " ~ typeof(option).stringof);
     }
+    version (DragonFlyBSD)
+    {
+        if (!hints.ai_socktype)
+        {
+            hints.ai_socktype = SOCK_STREAM;
+        }
+    }
 
     return () @trusted { return getAddressInfoImpl(node, service, &hints); }();
 }
```